core-api/src/main/java/com/optimizely/ab/bucketing/DecisionService.java

```diff
@@ -131,7 +131,13 @@ public DecisionService(@Nonnull Bucketer bucketer,
         if (ExperimentUtils.isUserInExperiment(projectConfig, experiment, filteredAttributes)) {
             String bucketingId = userId;
             if (filteredAttributes.containsKey(ControlAttribute.BUCKETING_ATTRIBUTE.toString())) {
-                bucketingId = filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString());
+                if (String.class.isInstance(filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString()))) {
+                    bucketingId = (String) filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString());
+                    logger.info("bucketingId is valid");
```

Comment:
nit: would be useful to print out what that bucketing ID is. Also, this should be a `debug` message.